lib/ClimaCoreTempestRemap/src/onlineremap.jl

```diff
@@ -68,6 +70,83 @@ function remap!(
     return target
 end
 
+# This version of this function is used for distributed remapping
+function remap!(
+    target::Fields.Field,
+    R::LinearMap,
+    source::Fields.Field,
+    context::ClimaComms.AbstractCommsContext,
+    target_mesh::Meshes.AbstractMesh{2},
+    space_o,
+)
+    pid = ClimaComms.mypid(context)
+    nprocs = ClimaComms.nprocs(context)
+
+    target_array = parent(target)
+    source_array = parent(source)
+
+    # Create mapping from global to local xelement indices (for distributed runs)
+    target_elemorder = Meshes.elements(target_mesh)
+    target_elempid = nothing
+    # To make IO easier, we enforce the partitioning to be sequential in the element ordering.
+    target_elempid, target_ranges =
+        Topologies.simple_partition(length(target_elemorder), nprocs)
+    @assert issorted(target_elempid)
+
+    target_local_elem_gidx = target_ranges[pid] # gidx = local_elem_gidx[lidx]
+    target_global_elem_lidx = Dict{Int, Int}() # inverse of local_elem_gidx: lidx = global_elem_lidx[gidx]
+    for (lidx, gidx) in enumerate(target_local_elem_gidx)
+        target_global_elem_lidx[gidx] = lidx
+    end
```

Comment:
1. Don't recompute this: use the information: use the `local_elem_gidx` from the topology
2. Store the dictionary in the `remap` object (or store it in the topology)

Reply:
Addressed in updated version of code - `target_global_elem_lidx` is created in `generate_map` using `topology.local_elem_gidx`, and is stored in the returned `LinearMap`